src/plugins/dist_matrixops/dist_identity.cpp

```diff
@@ -0,0 +1,310 @@
+#include <phylanx/execution_tree/primitives/node_data_helpers.hpp>
+#include <phylanx/execution_tree/tiling_annotations.hpp>
+#include <phylanx/ir/node_data.hpp>
+#include <phylanx/plugins/dist_matrixops/dist_identity.hpp>
+#include <phylanx/plugins/dist_matrixops/tile_calculation_helper.hpp>
+
+#include <hpx/errors/throw_exception.hpp>
+#include <hpx/include/lcos.hpp>
+#include <hpx/include/naming.hpp>
+#include <hpx/include/util.hpp>
+
+#include <array>
+#include <atomic>
+#include <cstddef>
+#include <cstdint>
+#include <memory>
+#include <string>
+#include <tuple>
+#include <utility>
+#include <vector>
+
+#include <blaze/Math.h>
+#include <blaze_tensor/Math.h>
+
+///////////////////////////////////////////////////////////////////////////////
+namespace phylanx { namespace dist_matrixops { namespace primitives {
+    ///////////////////////////////////////////////////////////////////////////
+    execution_tree::match_pattern_type const dist_identity::match_data = {
+
+        hpx::util::make_tuple("identity_d", std::vector<std::string>{R"(
+                identity_d(
+                    _1_sz,
+                    __arg(_2_tile_index, nil),
+                    __arg(_3_numtiles, nil),
+                    __arg(_4_name, ""),
+                    __arg(_5_tiling_type, "sym"),
+                    __arg(_6_dtype, nil)
+
+                )
+            )"},
+            &create_dist_identity,
+            &execution_tree::create_primitive<dist_identity>, R"(
+            sz, dtype, tile_index, numtiles, name, tiling_type,
+            Args:
+                sz (int): the size of a created (n x n) matrix.
+                tile_index (int): the tile index we need to generate the
+                    constant array for. A non-negative integer.
+                numtiles (int): number of tiles of the returned array
+                name (string, optional): the array given name. If not given, a
+                    globally unique name will be generated.
+                tiling_type (string, optional): defaults to `sym` which is a
+                    balanced way of tiling among all the numtiles localities.
+                    Other options are `row` or `column` tiling.
+                dtype (string, optional): the data-type of the returned array,
+                    defaults to 'float'.
+
+            Returns:
+
+            A part of an array of size 'size', which has the tile index of
+             'tile_index'.)")};
+
+    ///////////////////////////////////////////////////////////////////////////
+    dist_identity::dist_identity(
+        execution_tree::primitive_arguments_type&& operands,
+        std::string const& name, std::string const& codename)
+      : primitive_component_base(std::move(operands), name, codename)
+    {
+    }
+
+    ///////////////////////////////////////////////////////////////////////////
+    namespace detail {
+        static std::atomic<std::size_t> const_count(0);
```

Comment:
@hkaiser , as I think the namespace detail is common for `dist_identity` and other primitives in dist_matrixops, I have chosen different names for this variable in `dist_random` and `dist_constant`. This, `const_count`, is the same as `dist_constant`'s. Does it make a problem?

Reply:
Changed to identity_count.